### PR TITLE
Reduce memory usage of Analyzer test suite

### PR DESCRIPTION
**Pull request by hhobson**
## Change Description

Reduce memory usage of Presidio Analyzer test suite (excluding Azure AI language recognizer tests) down to ~6GB from ~8GB. When run locally the tests runtime went from ~60 seconds to ~25 seconds.

#### Before
Test suite memory usage before

#### After
Test suite memory usage after

### Changes

- Removed duplicate `nlp.load` call on each iteration of Stanza and transformers recognizer tests.
- Paramatized `test_when_analyze_with_unsupported_language_must_match` tests to avoid calling `AnalyzerEngine` five times in single test
- Mocked nlp engine in `test_when_analyze_with_unsupported_language_must_match` as test doesn't require nlp model to be loaded
- Removed unused fixtures - [pytest-unused-fixtures](https://github.com/mikicz/pytest-unused-fixtures) plugin used to detect them

## Issue reference

This PR fixes issue #1427

## Checklist

- [x] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [x] I have signed the CLA (if required)
- [x] My code includes unit tests
- [x] All unit tests and lint checks pass locally
- [ ] My PR contains documentation updates / additions if required

### CODE REVIEW

**Comment by hhobson** [comment]
@microsoft-github-policy-service agree

**Comment by omri374** [comment]
Thanks!

File: presidio-analyzer/tests/test_transformers_recognizer.py [hunk 1]
```diff
@@ -23,13 +23,6 @@ def nlp_engine(nlp_engines):
     return nlp_engine
 
 
-def prepare_and_analyze(nlp, recognizer, text, entities):
```
Comment (omri374) [inline presidio-analyzer/tests/test_transformers_recognizer.py:26]: Does removing this have an impact on memory usage?

Comment (hhobson) [reply presidio-analyzer/tests/test_transformers_recognizer.py:26]: Sorry for delay in responding @omri374, I missed your comment.

The specific impact on memory usage is the redundant `nlp.load()` call in the `prepare_and_analyze` function. `load()` is already called in the `nlp_engine` fixture.

Happy to revert back to using the `prepare_and_analyze` function, minus the load call if you'd prefer

Comment (omri374) [reply presidio-analyzer/tests/test_transformers_recognizer.py:26]: That's be great. Thanks.

Comment (hhobson) [reply presidio-analyzer/tests/test_transformers_recognizer.py:26]: Done

**Review by omri374 — APPROVED** [review]

**Pull request merged**